tests/cpp/test_memory.cpp

```diff
@@ -255,21 +256,27 @@ class TMALdstTest : public TMATest,
     dtype = std::get<1>(GetParam());
     dim = std::get<2>(GetParam());
 
+    // TODO: test less-nice shapes, for example 128 + 1 instead of 128
```

Comment:
I guess this should be already working. But I still want to leave this for the future to check. I am trying to keep the mental effort for this PR limited to the design and implementation of the schedule.

Reply:
I eventually decided to do it now. And as expected, it is working.